CMakeLists.txt

```diff
@@ -155,7 +155,7 @@ target_include_directories(jakt_stage1
 )
 add_executable(Jakt::jakt_stage1 ALIAS jakt_stage1)
 apply_output_rules(jakt_stage1)
-if (MSVC)
+if (MSVC OR (DEFINED MSVC_VERSION AND CMAKE_CXX_COMPILER_FRONTEND_VARIANT STREQUAL "GNU"))
```

Comment:
you only used the new option in one place. 

What is this checking for?

clang.exe? clang-cl?

If it's checking for clang with GNU variant, then ... is MSVC even the right check here? Should this just check that we're targeting windows? `if (WIN32)` ?

Reply:
MSVC matches clang-cl. MSVC_VERSION and GNU are meant to match clang++ on Windows. The issue is that in both scenarios I had linker memory errors and this resolved them. We could `if (WIN32)` for now, but if anyone tries to build with mingw that might get in their way, which is why I was trying to be targeted with what this will catch.

Reply:
`if (WIN32 and NOT (MINGW OR MSYS2))`?

Reply:
I suspect that this will also fail if you use clang.exe and tell it to use `lld.link` as the linker though, right? this is a link.exe thing only?

Reply:
I _thought_ that clang++ and lld-link were what got me started down this rabbit hole, but it's entirely possible I was just doing something wrong. Learned a few things in the past couple days!

I'll get mingw going and test all combinations.

Reply:
So, AK isn't going to work with mingw in its current form. MinGW doesn't always use win32 threads, it depends which package you're using, and they're mostly posix unless you want to build your own. AK hard expects win32 threads to be available if it detects the OS as Windows. I don't know how much effort/attention should be put into this given MSVC and Clang work.

Reply:
> I don't know how much effort/attention should be put into this given MSVC and Clang work.

None :). From cmake-bootstrap.md

> On MSYS2, g++ may error out with a "string table overflow" error. In that case, re-configure the build directory with -DCMAKE_BUILD_TYPE=MinSizeRel to get the -Os flag. Do note that using WSL2 or clang directly on windows is a more supported build platform. Maintainers will be reluctant to merge runtime or jakttest patches for MSYS2 quirks to keep the number of supported platfoms under control.

If it turns out that MSYS2 doesn't work anymore, so be it. we can replace that information with a "welp lmao" comment instead.